src/rpc.ts

```diff
@@ -60,7 +60,16 @@ router.post('/', async (req, res) => {
         return rpcError(res, 'INVALID_PARAMS', id);
       }
 
-      if (verifyUpdate(params.email, params.address, params.subscriptions, params.signature)) {
+      if (
+        // Do not check `subscriptions` for requests coming from
+        // envelop-ui, signed by backend
+        verifyUpdate(
+          params.email,
+          params.address,
+          params.address && params.address.length > 0 ? params.subscriptions : [],
+          params.signature
+        )
+      ) {
```

Comment:
```suggestion
      const isSignedByUser = verifyUpdate(
          params.email,
          params.address,
          params.address && params.address.length > 0 ? params.subscriptions : [],
          params.signature
        )
      if (isSignedByUser) {
```

Reply:
Variable name is not correct, since `verifyUpdate` can also be signed by backend. Maybe use `isValidSignature`

Reply:
ok, my point is not putting functions inside `if` statement. `isValidSignature` is also good.